### ISSUE TITLE
FastQueue perf compared to SDK

### ISSUE THREAD

**Issue by yevhen** [Scooletz/QueueBatch#4]
How does FastQueue achieves "up to 20x" faster than SDK provided client? What is the secret sauce?

**Comment by Scooletz** [comment]
I have two implementations, the SDK and the fast one: https://github.com/Scooletz/QueueBatch/blob/develop/src/QueueBatch/Impl/Queues/FastCloudQueue.cs

My fast approach uses:
- my `Span` based xml parser: https://github.com/Scooletz/QueueBatch/blob/develop/src/QueueBatch/Impl/Queues/FastXmlAzureParser.cs
- memory pooling for obtaining the response
- a custom `HttpContent` for sending messages https://github.com/Scooletz/QueueBatch/blob/develop/src/QueueBatch/Impl/Queues/FastCloudQueue.cs#L145

So basically, no allocations, no encoding and no `XmlReader`. Benchmarking might be affected by distance to DC (so for long hops it won't be 20x), still, this non-alloc approach pays off a lot.

If this answers it, please close the issue @yevhen

**Comment by yevhen** [comment]
Yes. Thank you!
